Mathlib/NumberTheory/Divisors.lean

```diff
@@ -514,3 +520,65 @@ theorem disjoint_divisors_filter_isPrimePow {a b : ℕ} (hab : a.Coprime b) :
   exact hn.ne_one (Nat.eq_one_of_dvd_coprimes hab han hbn)
 
 end Nat
+
+namespace Int
+variable {z : ℤ} {x : ℤ × ℤ}
+
+local notation "natCast" => Nat.castEmbedding (R := ℤ)
+local notation "negNatCast" =>
+  Function.Embedding.trans Nat.castEmbedding (Equiv.toEmbedding (Equiv.neg ℤ))
+
+/-- Pairs of divisors of an integer as a finset.
+
+`z.divisorsAntidiag` is the finset of pairs `(a, b) : ℤ × ℤ` such that `a * b = z`.
+As a special case, `Int.divisorsAntidiag 0 = ∅`.
+
+O(|n|). Computed from `Nat.divisorsAntidiagonal`. -/
+def divisorsAntidiag : ℤ → Finset (ℤ × ℤ)
```

Comment:
```suggestion
O(|z|). Computed from `Nat.divisorsAntidiagonal`. -/
def divisorsAntidiag : (z : ℤ) → Finset (ℤ × ℤ)
```
otherwise the variable in the docstring doesn't exist.